### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by KingsleyIbe**
### Milestone changes

In this milestone, I implemented the following:
- Create forms to perform the following functions:
    -- Creates a Post on behalf of the current_user.
    -- Create a comment on behalf of the current_user.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    new_comment = current_user.comments.new(
+      text: comment_params,
+      user_id: current_user.id,
+      post_id: @post.id
```
Comment (rukundoeric) [inline app/controllers/comments_controller.rb:7]: - [ ] <b>REQUIRED:</b> When you say `new_comment = current_user.comments.new()` the user_id is already initialize, so you don't need to re-initialize it again as you did on line 6. meaning your code would simply be like this:
```rb
@comment = current_user.comments.new(params)
@comment.post_id = params[:post_id]
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    )
+    new_comment.update_comments_counter
```
Comment (rukundoeric) [inline app/controllers/comments_controller.rb:9]: - [ ] <b>REQUIRED:</b> If you call `update_comments_counter` here, even if saving comment fails, comment count will still be updated, which is not correct. Kindly move it after line 10, where were are sure that the comment was saved successfully.

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class LikesController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    new_like = current_user.likes.new(
+      user_id: current_user.id,
+      post_id: @post.id
+    )
```
Comment (rukundoeric) [inline app/controllers/likes_controller.rb:7]: - [ ] <b>REQUIRED:</b>Also, here. When you say `current_user.likes.new()` the user_id is already initialized, so you don't need to re-initialize it again as you did on line 5. meaning your code would simply be like this:

```rb
@post = Post.find(params[:post_id])
@like = current_user.likes.new(post_id: @post.id)
```

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    new_like.update_likes_counter
```
Comment (rukundoeric) [inline app/controllers/likes_controller.rb:8]: - [ ] <b>REQUIRED:</b>Again here, If you call `update_likes_counter` here, even if saving like fails, likes count will still be updated, which is not correct. Kindly move it after line 9, where were are sure that the like was saved successfully.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @KingsleyIbe, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 


## Required Changes ♻️
   - [ ] I could not see a feature to create a Post. As the project requires, please create a feature for creating posts.
   - [ ] Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -19,4 +16,4 @@ def create
   def comment_params
     params.require(:comment).permit(:text)[:text]
   end
-end
+end
```
Comment (Emmyn5600) [inline app/controllers/comments_controller.rb]: - [ ] Great work on implementing the suggestions requested by the previous reviewer. however, there are some issues when I clicked on the see all post button. See the attached images to fix the issues.

The first issue is being occurred whenever you click on the `see all post` button and the way you are trying to get a post using the user or author id doesn't work, Please fix it. 

image

The second issue, occurred in the common/_user file where you are trying to render the user with a photo, bio, and name which is not working too. Kindly fix it.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @KingsleyIbe 

Good job so far! I know you have worked hard on making this milestone. There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Required Changes :recycle:
- [ ] Kindly check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (KingsleyIbe) [reply app/controllers/comments_controller.rb]: Hello @Emmyn5600 

Thank you for finding time to review my work, however, all my effort to reach you on slack proves abortive as you are not responding to your chats.

The first changes you requested have been fixed, please find an attached video. https://vimeo.com/694933063/37b7b7807e and the second is a feature implemented in the last milestone which is not a requirement for this milestone. however, all changes will be fixed as this is not the final milestone for this project.

Also, I noticed your changes are quite different from what the initial reviewer suggested.

File: app/controllers/comments_controller.rb (hunk 5 above)
Comment (KingsleyIbe) [reply app/controllers/comments_controller.rb]: Hello @Emmyn5600 

Thank you for finding time to review my work, however, all my effort to reach you on slack proves abortive as you are not responding to your chats.

The first changes you requested have been fixed, please find an attached video. https://vimeo.com/694933063/37b7b7807e and the second is a feature implemented in the last milestone which is not a requirement for this milestone. however, all changes will be fixed as this is not the final milestone for this project.

Also, I noticed your changes are quite different from what the initial reviewer suggested.

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @KingsleyIbe,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**